### PR TITLE
Add deploy github action

### PR DESCRIPTION
**Pull request by erikh360**

### CODE REVIEW

File: .ci/tag-from-ref.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+tag="${1#refs/tags/}"
+
+if [ "${tag}"  = "${1}" ]; then
```
Comment (jerith) [inline .ci/tag-from-ref.sh]: You can delete the extra space before the `=`. Also, it's probably best to add a trailing newline to the file.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
```
Comment (bearnard) [inline .github/workflows/deploy.yml:12]: missing the checkout action?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Get the version
```
Comment (jerith) [inline .github/workflows/deploy.yml:14]: Do we need an explicit `checkout` action here?

**Review by jerith — COMMENTED** [review]
Looks reasonable. Maybe test by tagging a commit in this branch with a "prerelease" version or something before merging?

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        id: get_version
+        run: |
+          .ci/tag-from-ref.sh "${GITHUB_REF}"
+      - name: Docker Build & Push
+        uses: mr-smithers-excellent/docker-build-push@v2
+        with:
+          image: menconnect-jsbox
+          tag: ${{ steps.get_version.outputs.TAG }}
+          registry: hub.docker.com
```
Comment (bearnard) [inline .github/workflows/deploy.yml]: `docker.io`

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        run: |
+          .ci/tag-from-ref.sh "${GITHUB_REF}"
+      - name: Docker Build & Push
+        uses: mr-smithers-excellent/docker-build-push@v2
+        with:
+          image: menconnect-jsbox
+          tag: ${{ steps.get_version.outputs.TAG }}
+          registry: docker.io
+          username: praekeltorgdeploy
+          password: secrets.DOCKER_HUB_PASS
```
Comment (bearnard) [inline .github/workflows/deploy.yml]: `${{ secrets.DOCKER_HUB_PASS }}`

**Review by bearnard — APPROVED** [review]

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Get the version
+        id: get_version
+        run: |
+          .ci/tag-from-ref.sh "${GITHUB_REF}"
+      - name: Docker Build & Push
+        uses: mr-smithers-excellent/docker-build-push@v2
+        with:
+          image: menconnect-jsbox
```
Comment (bearnard) [inline .github/workflows/deploy.yml]: this should be `praekeltfoundation/menconnect-jsbox`

**Pull request merged**